_posts/2019-06-18-list-strategy.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: posts
+title: "Automatic execution strategy selection in Bazel 0.27"
+authors:
+  - ishikhman
+---
+**tl;dr**: Exciting news! After [0.27](https://blog.bazel.build/2019/06/17/bazel-0.27.0.html) release Bazel will *auto select* a suitable execution strategy if none is provided. Customization and strategy enforcement will still be possible and will be much easier to configure.
+
+## Before Bazel 0.27
+
+It was possible to configure Bazel's [execution strategy](https://docs.bazel.build/versions/master/user-manual.html#strategy-options) for a build via flags `--spawn_strategy=`, `--strategy=Mnemonic=` and `--strategy_regexp=Regex=`. 
+This mechanism was quite powerful and widely used, but had some drawbacks:
+
+- Bazel's defaults did not account for remote execution at all, therefore we had to make sure to set the execution strategy for every mnemonic manually.
+
+- Starlark rules providing a persistent worker had to ship a .bazelrc file that sets --strategy=Mnemonic=worker (e.g. look at [rules_scala](https://github.com/bazelbuild/rules_scala)). 
+
+- Bazel has hardcoded defaults for native action mnemonics. For example, Bazel will use sandboxing (if it's available) for all actions by default but has a hardcoded default to use persistent workers for Java compilation actions.
+
+- To configure fallback strategy one had to provide additional flag `--remote_local_fallback_strategy=`.
+
+### Example
+I want to configure a build to run remotely and fallback onto `local` strategy in case remote execution is not possible. The configuration would be something like:
```

Comment:
Grammar. Avoid first person here.

Reply:
Also, format the example differently (here and below) so that it stands out on the page - say with a gray box or quotation. This example is a great illustration - if people just skim the blog post, you want them to see the example.

Reply:
done